lib/FrontendTool/ReferenceDependencies.cpp

```diff
@@ -31,20 +32,33 @@
 #include "llvm/Support/YAMLParser.h"
 
 using namespace swift;
+using namespace referenceDependencyKeys;
+
+template <typename Decl_T>
+static bool hasSomePrivateAccessLevel(Decl_T *decl) {
+  return decl->hasAccess() && decl->getFormalAccess() <= AccessLevel::FilePrivate;
+}
+
+/// If this declaration's privacy confines it to a file, no other file can
+/// depend upon it.
+template <typename Decl_T>
+static bool isVisibleOutside(Decl_T *decl) {
+  return !hasSomePrivateAccessLevel(decl);
+}
+
+template <typename Decl_T>
+static bool hasVisibleName(Decl_T *decl) {
```

Comment:
You can make all of these explicitly `const Decl_T *` too.

Reply:
Great! Have constified lots.